### PR TITLE
Implemented conda recipe for Rigetti's grove package

### PR DESCRIPTION
**Pull request by ntezak**
This implements a recipe for our open source library of quantum algorithms https://github.com/rigetticomputing/grove

### CODE REVIEW

File: recipes/grove/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    - python
+    - pip
+    # if your project compiles code (such as a C extension) then add `toolchain` as a build requirement.
+  run:
+    - python
+    - numpy
+    - qutip
+    - pyquil
+    - networkx
+    - pytest
```
Comment (synapticarbors) [inline recipes/grove/meta.yaml]: Is `pytest` actually required to run the package or just for testing. If it's the latter, please remove it. Same thing with `tox` and `mock`.

File: recipes/grove/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,97 @@
+# Note: there are many handy hints in comments in this example -- remove them when you've finalized your recipe
```
[the remark this answers is not in the document]
Comment (ntezak) [reply recipes/grove/meta.yaml]: Will do

File: recipes/grove/.condarc [hunk 3]
```diff
@@ -0,0 +1,6 @@
+channels:
```
[the remark this answers is not in the document]
Comment (ntezak) [reply recipes/grove/.condarc]: I hear this and can certainly understand, but the possibility to link cvxpy as a dependency is one of our primary motivations for using `conda` over plain old `pip` in the first place. Is there any chance we could add `cvxgrp` as a channel?

cc @mpharrigan

File: recipes/grove/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+requirements:
+  build:
+    - python
+    - pip
+    # if your project compiles code (such as a C extension) then add `toolchain` as a build requirement.
+  run:
+    - python
+    - numpy
+    - qutip
+    - pyquil
```
[the remark this answers is not in the document]
Comment (ntezak) [reply recipes/grove/meta.yaml]: No, with conda the version pinning should no longer be necessary. With pip the quoted dependencies are the only ones that successfully build on all platforms we try to support

File: recipes/grove/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+  # The pyQuil setup.py includes superfluous requirements
+  patches:
+    - remove-setup-requirements.patch
+
+build:
+  # Uncomment the following line if the package is pure python and the recipe is exactly the same for all platforms.
+  # It is okay if the dependencies are not built for all platforms/versions, although selectors are still not allowed.
+  # See https://conda-forge.org/docs/meta.html#building-noarch-packages for more details.
+  # noarch: python
+  number: 0
```
[the remark this answers is not in the document]
Comment (ntezak) [reply recipes/grove/meta.yaml]: Yes

**Comment by ntezak** [comment]
@synapticarbors Thank you for your review. If allowing `cvxgrp` as an external channel is a no-go, we can probably instruct our users to install it manually.

File: recipes/grove/.condarc (hunk 3 above)
[the remark this answers is not in the document]
Comment (mpharrigan) [reply recipes/grove/.condarc]: I can't imagine we're the only package that depends on `cvxpy` (??). How do others handle it? Should we encourage them to contribute their recipes to conda-forge

**Comment by ntezak** [comment]
Ping @synapticarbors just in case this PR has fallen off the radar.

**Comment by synapticarbors** [comment]
ping @conda-forge/staged-recipes -- This PR needs additional channels beyond `conda-forge` and `defaults`. I'm looking for thoughts on how best to handle this. My initial thought is that as a prerequisite, all packages that are currently getting pulled in via other channels need to be ported to conda-forge.

**Comment by isuruf** [comment]
All packages should be in `conda-forge` or `defaults`.  Which dependency is missing?

**Comment by ntezak** [comment]
Thanks, @synapticarbors and @isuruf for your feedback. The most complicated dependency of our package is `cvxpy` and its dependency `scs` which are nontrivial to build. We have therefore already made them optional dependencies for our package.

Unfortunately, it is beyond the scope of our work on this recipe to try and come up with a working conda recipe for the `cvxpy` stack, though I can certainly see the value of doing that.

Could we therefore proceed with this PR and add this recipe to conda-forge?

**Comment by synapticarbors** [comment]
If the stuff from `cvxpy` is optional, then it's fine to leave it off as a dependency and then just guide the user to install it if they need it. That's what happens for a lot of recipes that, for example work with various async frameworks (gevent, asyncio, twisted, etc), but are setup in a way that all don't need to be present to actually run.

**Comment by ntezak** [comment]
Great, @synapticarbors, then are there any changes you would like us to make to this current recipe?

**Comment by ntezak** [comment]
Thanks @synapticarbors. I made these changes and it seems to build ok!

**Comment by synapticarbors** [comment]
@ntezak -- Thanks for your contribution to conda-forge. A feedstock for this recipe should be generated soon and you'll be receiving several automated messages about it. When the feedstock has been rendered, please take a look at the README in that repo for information about updating and maintaining the recipe. One of the key points is all updates should be done by forking that repo and following a standard pull request workflow. If you haven't looked at it already the docs are quite helpful:

https://conda-forge.org/docs/

If you run into any difficulties please let us know.

**Pull request merged**